mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```
@@ -577,11 +846,29 @@ TuningParamSet *createTunableParamSpace(ModuleOp mod, TuningParamSetKind kind) {
   // create range and heuristic
   WalkResult findPrimary =
       mod->walk([&](rock::RockGemmWrapperInterface op) -> WalkResult {
+        GemmFeatures currentFeatures = rock::getFeatures(op);
+        // greedy is not implemented for non-accel
+        if (!rock::isAccel(currentFeatures) &&
+            kind == TuningParamSetKind::Greedy) {
+          kind = TuningParamSetKind::Exhaustive;
```

I think we should fail, instead of silently fallback to exhaustive? Better to let the user know that greedy cannot be used with non-accel

---

Another thing: We don't support Gemm greedy for non-accel, okay. But then why do we support GemmGemm? In other words, this check is not in `findGemmGemm`.

---

> I think we should fail, instead of silently fallback to exhaustive? Better to let the user know that greedy cannot be used with non-accel

we could do that, but then we have to split tier1-* files into f32 and non-f32 (and any file we use to tune that might have f32 configs). IMO it's not worth it. What do you guys think?

---

> Another thing: We don't support Gemm greedy for non-accel, okay. But then why do we support GemmGemm? In other words, this check is not in `findGemmGemm`.

because there's no non-accel support for gemm+gemm/attention.

---

> we could do that, but then we have to split tier1-* files into f32 and non-f32 (and any file we use to tune that might have f32 configs). IMO it's not worth it. What do you guys think?

Yes that is valid concern. Primary usecase of greedy is for development purpose. I don't expect it to be used inside CI or when generating quick tuning list for now.  We should print some log message warning about converting greedy to exhaustive though.

---

> We should print some log message warning about converting greedy to exhaustive though.

If we print anything it would break tuningRunner.py, because it parses the output of tuning-driver.

---

> > We should print some log message warning about converting greedy to exhaustive though.
> 
> If we print anything it would break tuningRunner.py, because it parses the output of tuning-driver.

Not if we print to stderr

---

that's a good point, I can do it. But tuningRunner will still hide this message to the user.

---

Yes, I think that's tuningRunner's fault and I really dislike it. I can solve that problem in the cleanup/refactoring PR I opened today: https://github.com/ROCm/rocMLIR/pull/2138